src/main/scala/is/hail/utils/package.scala

```diff
@@ -483,4 +483,19 @@ package object utils extends Logging
     finally
       resourceStream.close()
   }
+
+  def roundWithConstantSum(a: Array[Double]): Array[Int] = {
+    val withFloors = a.zipWithIndex.map { case (d, i) => (i, d, math.floor(d)) }
+    val totalFractional = (withFloors.map { case (i, orig, floor) => orig - floor }.sum + 0.5).toInt
```

Comment:
I think we should generalize that code.  I think we just need an efficient, general sortedIndex and some cleanup.  I'm happy to leave this for now and keep that as a separate task.